### PR TITLE
Scripting of Aluve App

### PR DESCRIPTION
**Pull request by MlulekiN**

### CODE REVIEW

File: Feature/Room.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Feature: Validating Rooms are displayed when date is input
+
+  Scenario Outline: The rooms are visible when user inputs date
+    Given The user is at the home page
```
Comment (benedictnkosi) [inline Feature/Room.feature]: on the homepage

File: Feature/Room.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    When The user clicks on the "Rooms" top menu
+    And The user clicks on the Click to Select Dates tab
```
Comment (benedictnkosi) [inline Feature/Room.feature]: please research term before using them. this is a field not a tab

File: Feature/Room.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Then User validates calendar is present
```
Comment (benedictnkosi) [inline Feature/Room.feature]: calendar is visible

File: src/test/java/Elements/RoomDisplayed_Elements.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package Elements;
+
+import org.openqa.selenium.By;
+
+public class RoomDisplayed_Elements {
+    public static By inputDate = By.xpath("//div[@class='hny-frm_grid']/child::span/input[@id='rooms_checkindate']");
+    public static By calendarTable = By.xpath("//div[@class='calendar-table']");
+    public static By todayDate = By.xpath("//div[@class='calendar-table']/descendant::td[contains(@class,'today')]");
+    public static By futureDate = By.xpath("//div[@class='calendar-table']/descendant::td[contains(@class,'available')]");
+    public static By whatsAppPage = By.xpath("//span/a[@data-testid='whatsapp_www_header_logo_link']");
```
Comment (benedictnkosi) [inline src/test/java/Elements/RoomDisplayed_Elements.java]: whats app - remove this

File: src/test/java/Elements/RoomDisplayed_Elements.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    public static By invalidDate = By.xpath("(//div[@class='calendar-table']//td[contains(@class,'disabled')])[1]");
+
+    public static By topColumn(String topColumn) {
```
Comment (benedictnkosi) [inline src/test/java/Elements/RoomDisplayed_Elements.java]: what is top column

File: src/test/java/Elements/VariousElements.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package Elements;
+
+import io.cucumber.datatable.DataTable;
+import org.openqa.selenium.By;
+
+public class VariousElements {
```
Comment (benedictnkosi) [inline src/test/java/Elements/VariousElements.java]: create a class for each page and put elements per page. also create a class for the menu and one class for the footer

File: src/test/java/Elements/VariousElements.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    public static By previousMonth = By.xpath("//div[@class='drp-calendar left']//table/descendant::th[@class='prev available']");
+    public static By nextMonth = By.xpath("//div[@class='drp-calendar right']//table/descendant::th[@class='next available']");
+    public static By roomsInputDate = By.xpath("//div[@class='hny-frm_grid']/child::span/descendant::input[@id='rooms_checkindate']");
+    public static By mapVisible = By.xpath("//div[contains(@class,'footer')]/descendant::iframe");
+    public static By selectRoom = By.xpath("//div[@class='booking-form-content']/descendant::div[@class='div-select-room-name']");
+    public static By bookNowButton = By.xpath("//button[text()='Book Now']");
+    public static By roomSelectBookNowButton = By.xpath("//div[@class='border-line']/descendant::a[text()='Book Now']");
+    public static By backToRoomsColumn = By.xpath("//div[@class='border-line']/descendant::a/following::a[@class='back']");
```
Comment (benedictnkosi) [inline src/test/java/Elements/VariousElements.java]: what is column

File: src/test/java/Elements/VariousElements.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    public static By displaySwitchPreviousButton = By.xpath("//div[@class='owl-nav']/button[@class='owl-prev']");
+    public static By displaySwitchNextButton = By.xpath("//div[@class='owl-nav']/button[@class='owl-next']");
+    public static By displayBookNowButton = By.xpath("//div[@class='banner-info-bg']/child::a[text()=' BOOK NOW']");
+    public static By PageNotFound = By.xpath("//h1[text()='Not Found']");
+    public static By inputTab(String inputTab) {
```
Comment (benedictnkosi) [inline src/test/java/Elements/VariousElements.java]: not called tab

File: src/test/java/Elements/VariousElements.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        return By.xpath("//div[@class='hny-frm_grid']/child::h5/following::input[contains(@placeholder,'" + inputTab + "')]");
+    }
+    public static By selectDate(DataTable dateStart) {
```
Comment (benedictnkosi) [inline src/test/java/Elements/VariousElements.java]: what is DataTAble?

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        testContext = context;
+        driver = testContext.getWebDriverManager().getDriver();
+        configReader = testContext.getConfigFileReader();
+        pageObjectManager = testContext.getPageObjectManager();
+
+    }
+
+    @Given("The user is at the home page")
+    public void the_user_has_logged_in_the_website() {
+        driver.get("https://dev-aluvegh.co.za/index.html");
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: read url from a properties file

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    @And("The user clicks on the Click to Select Dates tab")
+    public void theUserClicksOnTheClickToSelectDatesTab() {
+        WebElement dateSelect = driver.findElement(RoomDisplayed_Elements.inputDate);
+        dateSelect.click();
+    }
+
+    @Then("User validates calendar is present")
+    public void userValidatesCalendarIsPresent() {
+        WebElement calendar = driver.findElement(RoomDisplayed_Elements.calendarTable);
+        calendar.isDisplayed();
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: please use an junit Assert class for all validation steps

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    @When("User selects the {string} and {string}")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: the step name is not readable 
User selects the check in date {string} and check out date {string}

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+    public void userSelectsTheStartDateStartdateAndEndDateEnddate(String arg0, String arg1) {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: rename your args to something meaningful

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+        WebDriverWait wait = new WebDriverWait(driver, ofSeconds(5));
+        WebElement findDate1 = wait.until(ExpectedConditions.elementToBeClickable(RoomDisplayed_Elements.dateVariable(arg0)));
+        findDate1.click();
+
+        WebElement findDate2 = wait.until(ExpectedConditions.elementToBeClickable(RoomDisplayed_Elements.dateVariable(arg1)));
+        findDate2.click();
+    }
+
+    @And("The user clicks the {string} button")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: not readable

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+    public void theUserClicksTheApplyButton(String str) throws InterruptedException {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        try {
+            WebElement applyButton = wait.until(ExpectedConditions.elementToBeClickable(RoomDisplayed_Elements.actionButton(str)));
+            applyButton.click();
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: I created a click method for you please use that for all clicks

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    @Then("User validates the dates on input tab are start date and end date")
+    public void userValidatesTheDatesOnInputTabAreStartDateAndEndDate() {
+        WebElement dateSelect = driver.findElement(RoomDisplayed_Elements.inputDate);
+        String typeOut = dateSelect.toString().toLowerCase();
+        System.out.println("The date is " + typeOut);
+    }
+
+    @Then("User validates that {string} page option is visible")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: not readable

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+        ArrayList<String> tabs2 = new ArrayList<String>(driver.getWindowHandles());
+        driver.switchTo().window(tabs2.get(0));
+        driver.switchTo().window(tabs2.get(1));
+        WebElement whatsAppVisible = driver.findElement(RoomDisplayed_Elements.whatsAppPage);
+        whatsAppVisible.isDisplayed();
+        driver.close();
+        driver.switchTo().window(tabs2.get(0));
+    }
+
+    @Then("User validates if envelope page is open")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: what is envelope page

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+        System.out.println("The date is " + typeOut);
+    }
+
+    @Then("User validates that {string} page option is visible")
+    public void userValidatesThatWhatsappPageOptionIsVisible(String str) {
+        ArrayList<String> tabs2 = new ArrayList<String>(driver.getWindowHandles());
+        driver.switchTo().window(tabs2.get(0));
+        driver.switchTo().window(tabs2.get(1));
+        WebElement whatsAppVisible = driver.findElement(RoomDisplayed_Elements.whatsAppPage);
+        whatsAppVisible.isDisplayed();
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: not a rooms functionality move step to another class for footer

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+        driver.navigate().back();
+    }
+
+    @When("User clicks on invalid date")
+    public void userClicksOnInvalidDate() {
+        WebElement unselected = driver.findElement(RoomDisplayed_Elements.invalidDate);
+        Assert.assertFalse(unselected.isSelected());
+    }
+
+    @Then("User validates invalid cannot be selected")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: invalid what?

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+            WebElement applyButton = wait.until(ExpectedConditions.elementToBeClickable(RoomDisplayed_Elements.actionButton(str)));
+            applyButton.click();
+        } catch (Exception e) {
+            Thread.sleep(10);
+            WebElement applyButton = wait.until(ExpectedConditions.elementToBeClickable(RoomDisplayed_Elements.actionButton(str)));
+            applyButton.click();
+        }
+    }
+
+    @Then("User validates the dates on input tab are start date and end date")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: tab?

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+    public void userValidatesTheDatesOnInputTabAreStartDateAndEndDate() {
+        WebElement dateSelect = driver.findElement(RoomDisplayed_Elements.inputDate);
+        String typeOut = dateSelect.toString().toLowerCase();
+        System.out.println("The date is " + typeOut);
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: where is the validation here

File: Feature/Room.feature (hunk 1 above)
Comment (MlulekiN) [reply Feature/Room.feature]: Done

File: Feature/Room.feature (hunk 2 above)
Comment (MlulekiN) [reply Feature/Room.feature]: Done

File: Feature/Room.feature (hunk 3 above)
Comment (MlulekiN) [reply Feature/Room.feature]: Done

File: src/test/java/Elements/RoomDisplayed_Elements.java (hunk 4 above)
Comment (MlulekiN) [reply src/test/java/Elements/RoomDisplayed_Elements.java]: Done

File: src/test/java/Elements/RoomDisplayed_Elements.java (hunk 5 above)
Comment (MlulekiN) [reply src/test/java/Elements/RoomDisplayed_Elements.java]: Understood

File: src/test/java/Elements/VariousElements.java (hunk 7 above)
Comment (MlulekiN) [reply src/test/java/Elements/VariousElements.java]: Done

File: src/test/java/Elements/VariousElements.java (hunk 8 above)
Comment (MlulekiN) [reply src/test/java/Elements/VariousElements.java]: Done

File: src/test/java/Elements/VariousElements.java (hunk 6 above)
Comment (MlulekiN) [reply src/test/java/Elements/VariousElements.java]: Will do so...

File: src/test/java/Elements/VariousElements.java (hunk 9 above)
Comment (MlulekiN) [reply src/test/java/Elements/VariousElements.java]: I didn't know how to work with examples before but now it is sorted

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 10 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 12 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 14 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 15 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 17 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Removed. Footer class

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 18 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 20 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 21 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 16 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/Elements/RoomDisplayed_Elements.java [hunk 22]
```diff
@@ -0,0 +1,21 @@
+package Elements;
+
+import org.openqa.selenium.By;
+
+public class RoomDisplayed_Elements {
```
Comment (benedictnkosi) [inline src/test/java/Elements/RoomDisplayed_Elements.java]: elelements are created per page not per test scenario. call this Room Elements

File: src/test/java/Elements/RoomDisplayed_Elements.java [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+    public static By inputDate = By.xpath("//div[@class='hny-frm_grid']/child::span/input[@id='rooms_checkindate']");
+    public static By calendarTable = By.xpath("//div[@class='calendar-table']");
+    public static By invalidDate = By.xpath("(//div[@class='calendar-table']//td[contains(@class,'disabled')])[1]");
+
+    public static By topMenu(String topMenu) {
```
Comment (benedictnkosi) [inline src/test/java/Elements/RoomDisplayed_Elements.java]: move this to the menu page

File: src/test/java/Elements/RoomDisplayed_Elements.java [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+        return By.xpath("//div/descendant::li[contains(@class,'nav-item')]/child::a[text()='" + topMenu + "']");
+    }
+
+    public static By actionButton(String button) {
```
Comment (benedictnkosi) [inline src/test/java/Elements/RoomDisplayed_Elements.java]: rename this

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 25]
```diff
@@ -0,0 +1,91 @@
...
+    public Rooms_Are_Displayed(TestContext context) {
+        testContext = context;
+        driver = testContext.getWebDriverManager().getDriver();
+        configReader = testContext.getConfigFileReader();
+        pageObjectManager = testContext.getPageObjectManager();
+        homePage = pageObjectManager.getHomePage();
+
+    }
+
+    @Given("The user is on the home page")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: this is not a rooms displayed step

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 26]
```diff
@@ -0,0 +1,91 @@
...
+    public void the_user_has_logged_in_the_website() {
+        homePage.OpenHomePage();
+    }
+
+    @When("The user clicks on the {string} top menu")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: this is not a rooms displayed step

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 27]
```diff
@@ -0,0 +1,91 @@
...
+    public void theUserClicksOnTheTopColumn(String column) {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: i said remove column

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 28]
```diff
@@ -0,0 +1,91 @@
...
+        WebElement roomColumn = driver.findElement(RoomDisplayed_Elements.topMenu(column));
+        roomColumn.click();
+    }
+
+    @And("The user clicks on the *Click to Select Dates* date picker")
+    public void theUserClicksOnTheClickToSelectDatesTab() {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: is this a tab?

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 29]
```diff
@@ -0,0 +1,91 @@
...
+        try {
+            SeleniumActions.click(RoomDisplayed_Elements.dateVariable(dateCheck), driver);
+            SeleniumActions.click(RoomDisplayed_Elements.dateVariable(dateCheck1), driver);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @And("The user clicks the calendar button {string}")
+    public void theUserClicksTheApplyButton(String str) throws InterruptedException {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: function name incorrect

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 30]
```diff
@@ -0,0 +1,91 @@
...
+        SeleniumActions.click(RoomDisplayed_Elements.actionButton(str), driver);
+    }
+
+    @Then("User validates the dates on the date picker are check-in and check-out date")
+    public void userValidatesTheDatesOnInputTabAreStartDateAndEndDate() {
+        WebElement dateSelect = driver.findElement(RoomDisplayed_Elements.inputDate);
+        Assert.assertTrue(dateSelect.getText(), true);
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: what is this assert doing?

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 31]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @When("User clicks on invalid date")
+    public void userClicksOnInvalidDate() {
+        WebElement unselected = driver.findElement(RoomDisplayed_Elements.invalidDate);
+        Assert.assertFalse(unselected.isSelected());
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: is this an action or validation step?

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 32]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Then("User validates invalid cannot be selected")
+    public void userValidatesInvalidCannotBeSelected() {
+        Assert.assertFalse(String.valueOf(RoomDisplayed_Elements.invalidDate), false);
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: what is this assert doing?

File: src/test/java/Elements/RoomDisplayed_Elements.java [hunk 33]
```diff
@@ -0,0 +1,21 @@
+package Elements;
```
Comment (benedictnkosi) [inline src/test/java/Elements/RoomDisplayed_Elements.java]: remove this package, use the package of the page for all elements. each class must be the name of that page. e.g HomePage, RoomsPage

Comment (MlulekiN) [reply src/test/java/Elements/RoomDisplayed_Elements.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 31 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Fixed

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 30 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 25 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 26 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/Elements/RoomDisplayed_Elements.java (hunk 24 above)
Comment (MlulekiN) [reply src/test/java/Elements/RoomDisplayed_Elements.java]: Done

File: src/test/java/Elements/RoomDisplayed_Elements.java (hunk 22 above)
Comment (MlulekiN) [reply src/test/java/Elements/RoomDisplayed_Elements.java]: Done

File: src/test/java/Elements/RoomDisplayed_Elements.java (hunk 23 above)
Comment (MlulekiN) [reply src/test/java/Elements/RoomDisplayed_Elements.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 32 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/Elements/RoomDisplayed_Elements.java (hunk 22 above)
Comment (benedictnkosi) [reply src/test/java/Elements/RoomDisplayed_Elements.java]: rename

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 29 above)
Comment (benedictnkosi) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: function name still incorrect

File: Feature/Room.feature [hunk 34]
```diff
@@ -0,0 +1,17 @@
...
+
+  Scenario Outline: The rooms are visible when user inputs date
+    Given The user is on the home page
+    When The user clicks on the "Rooms" top menu
+    And The user clicks on the *Click to Select Dates* date picker
+    Then User validates calendar is visible
+
+    When User selects the check-in date "<start_date>" and check out date "<end_date>"
+    And The user clicks the calendar button 'Apply'
+    Then User checks if rooms are displayed
```
Comment (benedictnkosi) [inline Feature/Room.feature]: the user validates that....

File: Feature/Room.feature [hunk 35]
```diff
@@ -0,0 +1,17 @@
+Feature: Validating Rooms are displayed when date is input
+
+  Scenario Outline: The rooms are visible when user inputs date
+    Given The user is on the home page
+    When The user clicks on the "Rooms" top menu
+    And The user clicks on the *Click to Select Dates* date picker
```
Comment (benedictnkosi) [inline Feature/Room.feature]: add a validation step to check that you are on the Rooms page

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 36]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @And("The user clicks the calendar button {string}")
+    public void theUserClicksTheCalendarButton(String str) throws InterruptedException {
+        SeleniumActions.click(Room_Elements.calendarButton(str), driver);
+    }
+
+    @When("User clicks on invalid date")
+    public void userClicksOnInvalidDate() {
+        try {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: remove the try catches.

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 37]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Then("User validates calendar is visible")
+    public void userValidatesCalendarIsPresent() {
+        WebElement calendar = driver.findElement(Room_Elements.calendarTable);
+        Assert.assertTrue(calendar.isDisplayed());
+    }
+
+    @When("User selects the check-in date {string} and check out date {string}")
+    public void userSelectsTheStartDateStartdateAndEndDateEnddate(String dateCheck, String dateCheck1) {
+        try {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: remove the try catches.

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java [hunk 38]
```diff
@@ -0,0 +1,65 @@
...
+    WebDriver driver;
+
+    public Rooms_Are_Displayed(TestContext context) {
+        testContext = context;
+        driver = testContext.getWebDriverManager().getDriver();
+    }
+
+    @And("The user clicks on the *Click to Select Dates* date picker")
+    public void theUserClicksOnTheClickToSelectDatesTab() {
+        try {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/Rooms_Are_Displayed.java]: remove the try catches.

**Review by benedictnkosi — APPROVED** [review]

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 38 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 37 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: src/test/java/stepDefinition/Rooms_Are_Displayed.java (hunk 36 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/Rooms_Are_Displayed.java]: Done

File: Feature/Room.feature (hunk 35 above)
Comment (MlulekiN) [reply Feature/Room.feature]: Done

File: Feature/Room.feature (hunk 34 above)
Comment (MlulekiN) [reply Feature/Room.feature]: Done

**Pull request merged**